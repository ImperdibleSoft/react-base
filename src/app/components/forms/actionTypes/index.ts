import { FormActionType } from '../types';

export const FORM_RESET: FormActionType = 'FORM_RESET';
