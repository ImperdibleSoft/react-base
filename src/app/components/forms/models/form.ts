import { FormState } from '../types/form';

export const getState = (): FormState => ({
  canBeLocked: true,
  hasPermissions: false,
  isEditEnabled: false,
});
