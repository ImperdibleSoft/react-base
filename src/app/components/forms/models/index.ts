import { FieldTypes, PersistedSmartForm, SmartFormState } from '../types';

export const TEXT: FieldTypes = 'text';
export const EMAIL: FieldTypes = 'email';
export const NUMBER: FieldTypes = 'number';
export const PASSWORD: FieldTypes = 'password';
export const TEXTAREA: FieldTypes = 'textarea';
export const SWITCH: FieldTypes = 'switch';
export const CHECKBOX: FieldTypes = 'checkbox';
export const DROPDOWN: FieldTypes = 'dropdown';
export const DATE: FieldTypes = 'date';
export const BUTTON: FieldTypes = 'button';
export const SUBMIT: FieldTypes = 'submit';
export const ARRAY: FieldTypes = 'array';

const initForm = (): PersistedSmartForm => {
  return {
    state: 'pristine',
    errors: [],
  };
};

export const getState = (): SmartFormState => ({
  form: initForm(),
});
