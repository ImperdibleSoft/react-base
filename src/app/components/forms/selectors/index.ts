import { RootState } from '../../../types';

export const getFormState = ({ form }: RootState) => (formName: string): string =>
  form[formName] ? form[formName].state : 'pristine';

export const getFormErrors = ({ form }: RootState) => (formName: string): any[] =>
  form[formName] ? form[formName].errors : [];
