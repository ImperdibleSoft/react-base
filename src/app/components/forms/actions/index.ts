import { FORM_RESET } from '../actionTypes';

import { FormActionCreator } from '../types';

export const resetForm: FormActionCreator = payload => dispatch => {
  const { form } = payload;

  dispatch({
    type: FORM_RESET,
    payload: {
      form,
    },
  });
};
