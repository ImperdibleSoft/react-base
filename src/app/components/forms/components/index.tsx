import { Field as FormikField, FieldProps as FormikFieldProps } from 'formik';
import * as React from 'react';
import { log } from '../../../../common/utils/logger';
import analyticsApi from '../../../../common/apis/reports/analytics';

import { Buttons } from '../../touchable';
import Checkbox from './checkbox';
import DatePicker from './datepicker';
import Dropdown from './dropdown';
import Switch from './switch';
import Text from './text';

import { INTERACTIONS } from '../../../../common/constants/analytics';

import { TEXT, EMAIL, NUMBER, PASSWORD, TEXTAREA, SWITCH, CHECKBOX, DROPDOWN, DATE, BUTTON, SUBMIT } from '../models';

import { FieldProps } from '../types';

interface OwnProps {
  options: FieldProps;
}

const Field: React.FC<OwnProps> = ({ options }: OwnProps) => {
  const logSmartEvent = (action: string) => {
    const { icon, label, type } = options;

    analyticsApi.logSmartEvent({
      category: INTERACTIONS,
      label: `${label || icon} (${type} input)`,
      action,
    });
  };

  const onClick = (
    event: React.MouseEvent<HTMLInputElement | HTMLTextAreaElement | HTMLSelectElement | HTMLButtonElement>
  ) => {
    log(`onClick() of <${options.id}>`);
    logSmartEvent('Click');

    if (options.onClick) {
      options.onClick(event);
    }
  };

  const onFocus = (
    event: React.FocusEvent<HTMLInputElement | HTMLTextAreaElement | HTMLSelectElement | HTMLButtonElement>
  ) => {
    log(`onFocus() of <${options.id}>`);
    logSmartEvent('Focus');

    if (options.onFocus) {
      options.onFocus(event);
    }
  };

  const onChange = (
    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement | HTMLSelectElement | HTMLButtonElement>
  ) => {
    log(`onChange() of <${options.id}>`, options);
    logSmartEvent('Change');

    if (options.onChange) {
      options.onChange(event);
    }
  };

  interface RenderWrappedFieldProps {
    Component: typeof Text | typeof Switch | typeof Checkbox | typeof Dropdown | typeof DatePicker | typeof Buttons;
    props: any;
  }

  const renderWrappedField = ({ Component, props }: RenderWrappedFieldProps) => {
    const styles = {
      wrapper: {
        display: props.options.maxWidth ? 'inline-block' : 'block',
        maxWidth: props.options.maxWidth,
        verticalAlign: 'top',
      } as React.CSSProperties,
    };

    return (
      <span key={options.updateId || options.id} style={styles.wrapper}>
        <FormikField
          id={options.id}
          name={options.id}
          render={({ field: { value, onBlur, ...field } }: FormikFieldProps) => {
            const newProps = {
              ...props,
              ...field,
            };

            if (props.options.type === CHECKBOX || props.options.type === SWITCH) {
              newProps.options.isChecked = !!value;
            } else {
              newProps.options.model = value;
            }

            return (
              <Component
                {...newProps}
                onChange={(event: any) => {
                  props.onChange(event);
                  field.onChange(event);
                }}
              />
            );
          }}
        />
      </span>
    );
  };

  const parsedOptions = {
    ...options,
    isDisabled: (!options.isAlwaysEnabled && options.isDisabled) || options.isAlwaysDisabled,
  };

  switch (parsedOptions.type) {
    case TEXT:
    case EMAIL:
    case NUMBER:
    case PASSWORD:
    case TEXTAREA:
      return renderWrappedField({
        Component: Text,
        props: {
          onChange,
          onClick,
          onFocus,
          options: parsedOptions,
        },
      });

    case SWITCH:
      return renderWrappedField({
        Component: Switch,
        props: {
          onChange,
          onClick,
          onFocus,
          options: parsedOptions,
        },
      });

    case CHECKBOX:
      return renderWrappedField({
        Component: Checkbox,
        props: {
          onChange,
          onClick,
          onFocus,
          options: parsedOptions,
        },
      });

    case DROPDOWN:
      return renderWrappedField({
        Component: Dropdown,
        props: {
          onChange,
          onClick,
          onFocus,
          options: parsedOptions,
        },
      });

    case DATE:
      return renderWrappedField({
        Component: DatePicker,
        props: {
          onChange,
          onClick,
          onFocus,
          options: parsedOptions,
        },
      });

    case BUTTON:
    case SUBMIT:
      return renderWrappedField({
        Component: Buttons,
        props: {
          align: 'center',
          options: [parsedOptions],
        },
      });

    default:
      return null;
  }
};

export default Field;
