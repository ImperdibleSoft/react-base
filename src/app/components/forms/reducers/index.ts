import { error } from '../../../../common/utils/logger';

import { FORM_RESET } from '../actionTypes';

import { getState } from '../models';
import { PersistedSmartForm, SmartFormPayload, FormAction } from '../types';

const submitForm = (formState: PersistedSmartForm, payload: SmartFormPayload) => {
  if (!payload || !payload.form) {
    error('Dispatching an action form, without form property', payload);
    return formState;
  } else {
    return {
      ...formState,
      state: 'submitted',
      errors: [],
    };
  }
};

const succeedForm = (formState: PersistedSmartForm, payload: SmartFormPayload) => {
  if (!payload || !payload.form) {
    error('Dispatching an action form, without form property', payload);
    return formState;
  } else {
    return {
      ...formState,
      state: 'succeed',
      errors: [],
    };
  }
};

const parseErrors = (formState: PersistedSmartForm, payload: SmartFormPayload) => {
  if (!payload || !payload.form) {
    error('Dispatching an action form, without form property', payload);
    return formState;
  } else {
    return {
      ...formState,
      state: 'failed',
      errors: payload.errors,
    };
  }
};

const initialState = getState();

const reducer = (state = initialState, action: FormAction) => {
  const formAction = action as FormAction;

  // Any request
  if (/FORM_SUBMITTED/.test(formAction.type)) {
    return {
      ...state,
      [formAction.payload.form]: submitForm(state[formAction.payload.form], formAction.payload),
    };
  }

  // Any success
  if (/FORM_SUCCEED/.test(formAction.type)) {
    return {
      ...state,
      [formAction.payload.form]: succeedForm(state[formAction.payload.form], formAction.payload),
    };
  }

  // Any failure
  if (/FORM_FAILED/.test(formAction.type)) {
    return {
      ...state,
      [formAction.payload.form]: parseErrors(state[formAction.payload.form], formAction.payload),
    };
  }

  if (formAction.type === FORM_RESET) {
    return {
      ...state,
      [formAction.payload.form]: initialState[formAction.payload.form],
    };
  }

  return state;
};

export default reducer;
