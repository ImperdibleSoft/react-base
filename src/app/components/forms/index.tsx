import { Formik, FormikValues, FormikErrors } from 'formik';
import * as React from 'react';
import { useDispatch } from 'react-redux';

import { resetForm } from './actions';
import { updateFormStatus } from './actions/form';
import reducer from './reducers/form';

import Field from './components';
import { Buttons } from '../touchable';

import { CANCEL_ICON, EDIT_ICON, CANCEL_LABEL, EDIT_LABEL } from '../../../common/constants/buttons';
import { ButtonProps } from '../touchable/types';
import { BUTTON, CHECKBOX, SWITCH, SUBMIT } from './models';
import { getState } from './models/form';
import { FieldProps } from './types';

interface FormContent {
  Buttons: typeof Buttons;
  controlledButtons: ButtonProps[];
  controlledFields: FieldProps[];
  Field: typeof Field;
  isEditEnabled: boolean;
  values: FormikValues;
}

export const updateCollection = (originalCollection: FieldProps[], controlledCollection: FieldProps[]) =>
  originalCollection.map(field => {
    const controlledField = controlledCollection.find(f => f.id === field.id);
    return {
      ...field,
      ...controlledField,
      error: controlledField && controlledField.error ? controlledField.error : field.error,
      model: controlledField && controlledField.model ? controlledField.model : field.model,
    };
  });

const mapFieldValues = (field: FieldProps) => {
  const values: FormikValues = {};

  if (field.fields) {
    field.fields.forEach(f => {
      const fieldValues = mapFieldValues(f);
      Object.keys(fieldValues).forEach(key => {
        values[key] = fieldValues[key];
      });
    });
  } else if (field.type === CHECKBOX || field.type === SWITCH) {
    values[field.id] = field.isChecked;
  } else {
    values[field.id] = field.model;
  }

  return values;
};

const getInitialValues = (fields: FieldProps[]) => {
  const values: FormikValues = {};

  fields.forEach(field => {
    const fieldValues = mapFieldValues(field);
    Object.keys(fieldValues).forEach(key => {
      values[key] = fieldValues[key];
    });
  });

  return values;
};

const mapValuesToField = (field: FieldProps, values: FormikValues): FieldProps => {
  if (field.fields) {
    return {
      ...field,
      fields: field.fields.map(f => {
        return mapValuesToField(f, values);
      }),
    };
  }

  if (field.type === CHECKBOX || field.type === SWITCH) {
    return {
      ...field,
      error: undefined,
      isChecked: values[field.id],
    };
  }

  return {
    ...field,
    error: undefined,
    model: values[field.id],
  };
};

const mapValuesToFields = (fields: FieldProps[], values: FormikValues) =>
  fields.map(field => mapValuesToField(field, values));

interface OwnProps {
  buttons: ButtonProps[];
  children: (props: FormContent) => React.ReactNode;
  defaultEnabled?: boolean;
  fields: FieldProps[];
  form: string;
  handleSubmit: (fields: FieldProps[]) => void;
  isFetching: boolean;
  validate?: (values: FormikValues) => FormikErrors<FormikValues>;
}

const Form: React.FC<OwnProps> = ({
  buttons,
  children,
  defaultEnabled,
  fields,
  form,
  handleSubmit,
  isFetching,
  validate,
}: OwnProps) => {
  const [{ canBeLocked, hasPermissions, isEditEnabled }, stateDispatch] = React.useReducer(reducer, getState());

  const initialValues = getInitialValues(fields);
  const dispatch = useDispatch();

  React.useEffect(() => {
    // Reset form data
    dispatch(resetForm({ form }));

    // Set initial form status
    stateDispatch(
      updateFormStatus({
        canBeLocked: false,
        hasPermissions: true,
        isEditEnabled: defaultEnabled === true,
      })
    );
  }, []);

  const onSubmit = (values: FormikValues): void => {
    if (isEditEnabled && (!canBeLocked || (canBeLocked && hasPermissions))) {
      handleSubmit(mapValuesToFields(fields, values));
    }
  };

  const toggleEnabledState = () => {
    stateDispatch(
      updateFormStatus({
        isEditEnabled: !isEditEnabled,
      })
    );
  };

  const toggleButton: ButtonProps = {
    buttonType: isEditEnabled ? undefined : 'primary',
    icon: isEditEnabled ? CANCEL_ICON : EDIT_ICON,
    id: 'toggle',
    isDisabled: canBeLocked && !hasPermissions,
    label: isEditEnabled ? CANCEL_LABEL : EDIT_LABEL,
    onClick: toggleEnabledState,
    type: BUTTON,
  };

  return (
    <Formik initialValues={initialValues} validate={validate} onSubmit={onSubmit}>
      {({
        /* eslint-disable @typescript-eslint/no-unused-vars */
        errors,
        handleBlur,
        handleChange,
        handleSubmit: handleFormikSubmit,
        isSubmitting,
        touched,
        values,
        /* eslint-enable @typescript-eslint/no-unused-vars */
      }) => (
        <form id={form} name={form} onSubmit={handleFormikSubmit} noValidate>
          {children({
            Buttons,
            controlledButtons: buttons.map(b => ({
              ...b,
              isDisabled: b.type === SUBMIT || b.isDisabled || !isEditEnabled || isFetching,
            })),
            controlledFields: fields.map(f => ({
              ...f,
              isDisabled: f.isDisabled || !isEditEnabled || isFetching,
            })),
            Field,
            isEditEnabled,
            values,
          })}

          {canBeLocked && <Buttons options={[toggleButton]} />}
        </form>
      )}
    </Formik>
  );
};

export default Form;
