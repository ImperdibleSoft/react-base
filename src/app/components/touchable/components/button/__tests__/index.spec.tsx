import * as React from 'react';
import { shallow } from 'enzyme';

import Button from '..';
import { BUTTON, SUBMIT } from '../../../../forms/models';

describe('Button', () => {
  describe('by type', () => {
    it('should render a button', () => {
      const component = shallow(<Button options={{ id: 'button', label: 'Button', type: BUTTON }} />);
      expect(component).toMatchSnapshot();
    });

    it('should render a button with a text-icon', () => {
      const component = shallow(
        <Button options={{ id: 'button', label: 'Button', icon: 'send', iconLast: true, type: BUTTON }} />
      );
      expect(component).toMatchSnapshot();
    });

    it('should render a link that looks like a button', () => {
      const component = shallow(<Button options={{ id: 'button', label: 'Link', to: '/url', type: BUTTON }} />);
      expect(component).toMatchSnapshot();
    });

    it('should render a submit button', () => {
      const component = shallow(<Button options={{ id: 'button', label: 'Submit', type: SUBMIT }} />);
      expect(component).toMatchSnapshot();
    });
  });

  describe('by colors and shapes', () => {
    it('should render a danger button', () => {
      const component = shallow(
        <Button options={{ buttonType: 'danger', id: 'button', label: 'Button', type: BUTTON }} />
      );
      expect(component).toMatchSnapshot();
    });

    it('should render a disabled button', () => {
      const component = shallow(
        <Button options={{ buttonType: 'disabled', id: 'button', label: 'Button', type: BUTTON }} />
      );
      expect(component).toMatchSnapshot();
    });

    it('should render a facebook button', () => {
      const component = shallow(
        <Button options={{ buttonType: 'facebook', id: 'button', label: 'Button', type: BUTTON }} />
      );
      expect(component).toMatchSnapshot();
    });

    it('should render a google button', () => {
      const component = shallow(
        <Button options={{ buttonType: 'google', id: 'button', label: 'Button', type: BUTTON }} />
      );
      expect(component).toMatchSnapshot();
    });

    it('should render a instagram button', () => {
      const component = shallow(
        <Button options={{ buttonType: 'instagram', id: 'button', label: 'Button', type: BUTTON }} />
      );
      expect(component).toMatchSnapshot();
    });

    it('should render a linkedin button', () => {
      const component = shallow(
        <Button options={{ buttonType: 'linkedin', id: 'button', label: 'Button', type: BUTTON }} />
      );
      expect(component).toMatchSnapshot();
    });

    it('should render a primary button', () => {
      const component = shallow(
        <Button options={{ buttonType: 'primary', id: 'button', label: 'Button', type: BUTTON }} />
      );
      expect(component).toMatchSnapshot();
    });

    it('should render a success button', () => {
      const component = shallow(
        <Button options={{ buttonType: 'success', id: 'button', label: 'Button', type: BUTTON }} />
      );
      expect(component).toMatchSnapshot();
    });

    it('should render a twitter button', () => {
      const component = shallow(
        <Button options={{ buttonType: 'twitter', id: 'button', label: 'Button', type: BUTTON }} />
      );
      expect(component).toMatchSnapshot();
    });

    it('should render a transparent button', () => {
      const component = shallow(
        <Button options={{ isTransparent: true, id: 'button', label: 'Button', type: BUTTON }} />
      );
      expect(component).toMatchSnapshot();
    });

    it('should render a warning button', () => {
      const component = shallow(
        <Button options={{ buttonType: 'warning', id: 'button', label: 'Button', type: BUTTON }} />
      );
      expect(component).toMatchSnapshot();
    });
  });
});
