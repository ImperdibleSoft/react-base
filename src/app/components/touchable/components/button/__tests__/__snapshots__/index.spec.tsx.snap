// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button by colors and shapes should render a danger button 1`] = `
<ButtonAsButton
  buttonType="danger"
  disabled={true}
  id="button"
  isDisabled={true}
  onClick={[Function]}
  type="button"
>
  <Content
    options={
      Object {
        "icon": undefined,
        "iconLast": undefined,
        "label": "Button",
      }
    }
  />
</ButtonAsButton>
`;

exports[`Button by colors and shapes should render a disabled button 1`] = `
<ButtonAsButton
  buttonType="disabled"
  disabled={true}
  id="button"
  isDisabled={true}
  onClick={[Function]}
  type="button"
>
  <Content
    options={
      Object {
        "icon": undefined,
        "iconLast": undefined,
        "label": "Button",
      }
    }
  />
</ButtonAsButton>
`;

exports[`Button by colors and shapes should render a facebook button 1`] = `
<ButtonAsButton
  buttonType="facebook"
  disabled={true}
  id="button"
  isDisabled={true}
  onClick={[Function]}
  type="button"
>
  <Content
    options={
      Object {
        "icon": undefined,
        "iconLast": undefined,
        "label": "Button",
      }
    }
  />
</ButtonAsButton>
`;

exports[`Button by colors and shapes should render a google button 1`] = `
<ButtonAsButton
  buttonType="google"
  disabled={true}
  id="button"
  isDisabled={true}
  onClick={[Function]}
  type="button"
>
  <Content
    options={
      Object {
        "icon": undefined,
        "iconLast": undefined,
        "label": "Button",
      }
    }
  />
</ButtonAsButton>
`;

exports[`Button by colors and shapes should render a instagram button 1`] = `
<ButtonAsButton
  buttonType="instagram"
  disabled={true}
  id="button"
  isDisabled={true}
  onClick={[Function]}
  type="button"
>
  <Content
    options={
      Object {
        "icon": undefined,
        "iconLast": undefined,
        "label": "Button",
      }
    }
  />
</ButtonAsButton>
`;

exports[`Button by colors and shapes should render a linkedin button 1`] = `
<ButtonAsButton
  buttonType="linkedin"
  disabled={true}
  id="button"
  isDisabled={true}
  onClick={[Function]}
  type="button"
>
  <Content
    options={
      Object {
        "icon": undefined,
        "iconLast": undefined,
        "label": "Button",
      }
    }
  />
</ButtonAsButton>
`;

exports[`Button by colors and shapes should render a primary button 1`] = `
<ButtonAsButton
  buttonType="primary"
  disabled={true}
  id="button"
  isDisabled={true}
  onClick={[Function]}
  type="button"
>
  <Content
    options={
      Object {
        "icon": undefined,
        "iconLast": undefined,
        "label": "Button",
      }
    }
  />
</ButtonAsButton>
`;

exports[`Button by colors and shapes should render a success button 1`] = `
<ButtonAsButton
  buttonType="success"
  disabled={true}
  id="button"
  isDisabled={true}
  onClick={[Function]}
  type="button"
>
  <Content
    options={
      Object {
        "icon": undefined,
        "iconLast": undefined,
        "label": "Button",
      }
    }
  />
</ButtonAsButton>
`;

exports[`Button by colors and shapes should render a transparent button 1`] = `
<ButtonAsButton
  disabled={true}
  id="button"
  isDisabled={true}
  isTransparent={true}
  onClick={[Function]}
  type="button"
>
  <Content
    options={
      Object {
        "icon": undefined,
        "iconLast": undefined,
        "label": "Button",
      }
    }
  />
</ButtonAsButton>
`;

exports[`Button by colors and shapes should render a twitter button 1`] = `
<ButtonAsButton
  buttonType="twitter"
  disabled={true}
  id="button"
  isDisabled={true}
  onClick={[Function]}
  type="button"
>
  <Content
    options={
      Object {
        "icon": undefined,
        "iconLast": undefined,
        "label": "Button",
      }
    }
  />
</ButtonAsButton>
`;

exports[`Button by colors and shapes should render a warning button 1`] = `
<ButtonAsButton
  buttonType="warning"
  disabled={true}
  id="button"
  isDisabled={true}
  onClick={[Function]}
  type="button"
>
  <Content
    options={
      Object {
        "icon": undefined,
        "iconLast": undefined,
        "label": "Button",
      }
    }
  />
</ButtonAsButton>
`;

exports[`Button by type should render a button 1`] = `
<ButtonAsButton
  disabled={true}
  id="button"
  isDisabled={true}
  onClick={[Function]}
  type="button"
>
  <Content
    options={
      Object {
        "icon": undefined,
        "iconLast": undefined,
        "label": "Button",
      }
    }
  />
</ButtonAsButton>
`;

exports[`Button by type should render a button with a text-icon 1`] = `
<ButtonAsButton
  disabled={true}
  id="button"
  isDisabled={true}
  onClick={[Function]}
  type="button"
>
  <Content
    options={
      Object {
        "icon": "send",
        "iconLast": true,
        "label": "Button",
      }
    }
  />
</ButtonAsButton>
`;

exports[`Button by type should render a link that looks like a button 1`] = `
<LinkAsButton
  id="button"
  onClick={[Function]}
>
  <Link
    to="/url"
  >
    <Content
      options={
        Object {
          "icon": undefined,
          "iconLast": undefined,
          "label": "Link",
        }
      }
    />
  </Link>
</LinkAsButton>
`;

exports[`Button by type should render a submit button 1`] = `
<ButtonAsButton
  id="button"
  onClick={[Function]}
  type="submit"
>
  <Content
    options={
      Object {
        "icon": undefined,
        "iconLast": undefined,
        "label": "Submit",
      }
    }
  />
</ButtonAsButton>
`;
